docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+    * Alpha releases based on the main branch can be released when necessary
+* Week 13-14:
+    * Beta releases based on the main branch
+    * The beta releases are meant as a preview what the upcoming release will look like
+    * Provider can start adopting the new release based on the beta releases
+    * Breaking changes to providers have to be discussed and approved in the office hours
+* Week 15-16:
+    * The release branch is created
+    * RC releases based on the release branch
+    * There is a feature freeze on the release branch
+    * Only bug fixes / documentation / tests fixes and additions are allowed
+* Week 17:
+    * `.0` GA release is created based on the release branch
+* After that:
+    * `.1+`: Monthly patch releases based on the release branch
+    * Provider releases corresponding to the CAPI minor release when they are ready
+
+Some additional notes:
+
+* Support for new Kubernetes releases is first implemented on the main branch, then cherry-picked into
+  supported release branches when feasible and eventually released in the next monthly patch release(s).
+* New Kubernetes and Controller-Runtime (CR) minor releases are only picked up on the main branch.
+
+The following diagram visualizes the release cycles with the v1.2, v1.3 and v1.4 releases:
+
+<!-- The release cycle png can be opened and edited in draw.io -->
+![Release Cycle Overview](release-cycle-overview.png)
+
+# v1.3 Release Cycle timeline
+
+The following table shows the preliminary dates for the `v1.3` release cycle.
+
+| **What**                                       | **Who** | **When**                   | **Week** |
+|------------------------------------------------|---------|----------------------------|----------|
+| Start of Release Cycle                         | Lead    | Monday 18th July 2022      | week 1   |
+| 1.3.0-beta.0 released                          | Lead    | Tuesday 1st November 2022  | week 13  |
+| 1.3.0-beta.1 released                          | Lead    | Tuesday 8th November 2022  | week 14  |
+| release-1.3 branch created (**[Code Freeze]**) | Lead    | Tuesday 15th November 2022 | week 15  |
+| release-1.3 jobs created                       | Lead    | Tuesday 15th November 2022 | week 15  |
+| 1.3.0-rc.0 released                            | Lead    | Tuesday 15th November 2022 | week 15  |
+| 1.3.0-rc.1 released                            | Lead    | Tuesday 22th November 2022 | week 16  |
+| **1.3.0 released**                             | Lead    | Tuesday 29th November 2022 | week 17  |
+
+# v1.4 Release Cycle timeline
+
+The following table shows the preliminary dates for the `v1.4` release cycle.
+
+| **What**                                             | **Who** | **When**                   | **Week** |
+|------------------------------------------------------|---------|----------------------------|----------|
+| Start of Release Cycle                               | Lead    | Monday 5th December 2022   | week 1   |
```

Comment:
should this table also have patch release dates?

Reply:
I'm just worried that it looks like the release team has nothing to do between week 1 and week 14 which is not true since patch releases are a lot of work too

Reply:
Ah yup, that makes sense. I'll see if there's a good way to integrate it

Reply:
It was definitely a good idea to add the dates to the table as it allows aligning patch releases with beta/GA releases of the current cycle